pkg/log/zap.go

```diff
@@ -32,3 +34,18 @@ func NewDefaultZapLogger() (logr.Logger, error) {
 	logger := zapr.NewLogger(zapLogger)
 	return logger, nil
 }
+
+func NewZapLoggerWithWriter(out io.Writer) logr.Logger {
+
+	bWriter := out
+	config := zap.NewDevelopmentConfig()
+	config.EncoderConfig.EncodeLevel = zapcore.CapitalLevelEncoder
+	config.EncoderConfig.FunctionKey = "function"
+	
+	core := zapcore.NewCore(zapcore.NewJSONEncoder(config.EncoderConfig), zapcore.AddSync(bWriter), config.Level)
+	zapLogger := zap.New(core)
+	zapLogger.Error("an error")
```

Comment:
please remove this line of code

Reply:
@STRRL  Can you please elaborate on how should i create a logger if i removed these three lines ?
or i can use flags to create a logger like this 
`typeNewZapLoggerWithWriter {
	out  log.Logger
}
func NewZapLoggerWithWriter(w io.Writer) logr.Logger{
	return &simpleLogger{
		out: log.New(w, "", log.LstdFlags|log.Lmicroseconds),
	}
}


`

Reply:
please only remove L47 :)